lib/graphiti.rb

```diff
@@ -29,6 +29,14 @@ def self.context=(val)
   def self.with_context(obj, namespace = nil)
     prior = context
     self.context = {object: obj, namespace: namespace}
+    # conditional reads per request
+    resources.each do |resource_class|
+      resource_class.sideloads.values.each do |sideload|
+        flag = sideload.instance_variable_get(:@readable)
+        evaluated = sideload.send(:evaluate_flag, flag)
+        resource_class.serializer.relationship(sideload.name, if: -> { evaluated })
+      end
+    end
```

Comment:
After a lot of digging https://github.com/graphiti-api/graphiti/pull/187 put me on the right track...the above may be ugly but I can confirm it is a working solution (for reads)...so that's a step in the right direction at least.

The issue is that all resource serializer logic pertaining to sideloads is evaluated on class load and the request context has no affect thereafter. 

Interestingly this isn't the case for attributes so perhaps the germ of a tidy solution lies in how they are handled?

I wasn't able to write a test to catch this unfortunately. The one in this PR passes regardless of fix above.  To recreate this issue run a production server (cache_classes = true) and add a binding or logger in the conditional method governing the sideload read.

You will observe that it fires on class load only.

```ruby
class EmployeeResource < Graphiti::Resource
  has_many positions, readable: :admin?
end

class PositionResource < Graphiti::Resource
  def admin?
     Rails.logger.debug("readable? guard called")
     Context.current_user.admin?
  end
end
```